examples/tutorials/code_2_hyperband_checkpoint.py

```diff
@@ -0,0 +1,392 @@
+    )
+
+    trials = 1
+    while not experiment.is_done:
+        print("trial", trials)
+        trial = experiment.suggest()
+        if trial is None and experiment.is_done:
+            break
+        valid_error_rate = main(
+            **trial.params, checkpoint=f"{experiment.working_dir}/{trial.hash_params}"
+        )
+        experiment.observe(trial, valid_error_rate, name="valid_error_rate")
+        trials += 1
+
+
+#%%
+# Let's run the optimization now. You may want to reduce the maximum number of epochs in
+# ``fidelity(1, 120, base=4)`` and set the number of ``repetitions`` to 1 to get results more
+# quickly. With current configuration, this example takes 2 days to run on a Titan RTX.
+
+experiment = run_hpo()
+
+#%%
+# Analysis
+# --------
+#
+# That is all for the checkpointing example. We should nevertheless analyse the results
+# before wrapping up this tutorial.
+#
+# We should first look at the :ref:`sphx_glr_auto_examples_plot_1_regret.py`
+# to verify the optimization with Hyperband.
+
+fig = experiment.plot.regret()
+fig.show()
+
+#%%
+# .. This file is produced by docs/scripts/build_database_and_plots.py
+#
+# .. raw:: html
+#     :file: ../_static/hyperband-cifar10_regret.html
+#
+#
+# Moving the cursor over the points, we see that only a handful of trials
+# lead to better results with 1 epoch. Otherwise, all other trials with validation error rate
+# below 80% were trained for more than 1 epoch.
+# The best found result is high, a validation accuracy 23.6%. With VGG11 we could expect to achieve
+# lower than 10%. To see if the search space may be the issue, we first look at the
+# :ref:`sphx_glr_auto_examples_plot_3_lpi.py`.
+
+fig = experiment.plot.lpi()
+fig.show()
+
+#%%
+# .. raw:: html
+#     :file: ../_static/hyperband-cifar10_lpi.html
+#
+# The momentum and weight decay had very large priors, yet the different values
+# had no important effect on the validation accuracy. We can ignore them.
+# For the learning rate and for gamma
+# it is worth looking at the :ref:`sphx_glr_auto_examples_plot_4_partial_dependencies.py`
+# to see if the search space was perhaps too narrow or too large.
+
+fig = experiment.plot.partial_dependencies(params=["gamma", "learning_rate"])
+fig.show()
+
+# sphinx_gallery_thumbnail_path = '_static/restart.png'
+
+#%%
+# .. This file is produced by docs/scripts/build_database_and_plots.py
+#
+# .. raw:: html
+#     :file: ../_static/hyperband-cifar10_partial_dependencies_params.html
+#
+# The main culprit for the high validation error rate seems to be the wide prior for ``gamma``.
+# Because of this Hyperband spent most of the computation time on bad ``gamma``s. This prior
+# should be narrowed to ``uniform(0.995, 1)``.
+# The prior for the learning rate could also be narrowed to ``loguniform(0.001, 0.1)`` to
+# help the optimization.
+#
+# Note that Hyperband could also find better results without adjusting the search space, but
+# it would required significantly more repetitions.
```

Comment:
At end of rendered HTML file, the text "Total running time of the script: ( 0 minutes 0.000 seconds)" appears, but not sure what this is all about. It also appears in all of the visualization doc pages.

Reply:
I'll see if there is an option to remove this. It's part of the sphinx-gallery, when it executes the scripts.

Reply:
I could not find an option to remove it.